dlx_rest/api/api.py

```diff
@@ -0,0 +1,1917 @@
+class RecordsListCount(Resource):
+    @ns.expect(RecordsList.args)
+    def get(self, collection):
+        cls = ClassDispatch.batch_by_collection(collection) or abort(404)
+        args = RecordsList.args.parse_args()
+
+        if args.search:
+            search = unquote(args.search)
+            
+            try:
+                query = Query.from_string(search, record_type=collection[:-1])
+            except InvalidQueryString as e:
+                abort(422, str(e))
+
+        else:
+            query = {}
+
+        links = {
+            '_self': URL('api_records_list_count', collection=collection, search=args.search).to_str(),
+            'related': {
+                'records': URL('api_records_list', collection=collection, search=args.search).to_str()
+            }
+        }
+        
+        meta = {'name': 'api_records_list_count', 'returns': URL('api_schema', schema_name='api.count').to_str()}
+        
+        if query:
+            data = cls().handle.count_documents(
+                query.compile(),
+                # collation is not implemented in mongomock
+                collation=Collation(locale='en', strength=1, numericOrdering=True) if Config.TESTING == False else None,
+                maxTimeMS=Config.MAX_QUERY_TIME
+            )
+        else:
+            data = cls().handle.estimated_document_count()
+
+        return ApiResponse(links=links, meta=meta, data=data).jsonify()
+
+# Records list browse
+@ns.route('/marc/<string:collection>/records/browse')
+@ns.param('collection', '"bibs" or "auths"')
+class RecordsListBrowse(Resource):
+    args = reqparse.RequestParser()
+    args.add_argument(
+        'search',
+        type=str, 
+        help='Consult documentation for query syntax. The logical field to browse by must be the first search term'
+    )
+    args.add_argument(
+        'compare',
+        type=str,
+        choices=['greater', 'less'],
+        help='Return the results "greater than" or "less than" the match, with the matched field first or last last if there is one'
+    )
+    args.add_argument(
+        'start', 
+        type=int, 
+        help='Result to start list at',
+        default=1
+    )
+    args.add_argument(
+        'limit', 
+        type=int,
+        help='Number of results to return. Max is 100',
+        default=10,
+    )
+    args.add_argument(
+        'type',
+        type=str, 
+        choices=['bib', 'speech', 'vote', 'auth']
+    )
+    
+    @ns.doc(description='Return a list of MARC Bibliographic or Authority Records sorted by the "logical field" specified in the search.')
+    @ns.expect(args)
+    def get(self, collection):
+        args = RecordsListBrowse.args.parse_args()
+        cls = ClassDispatch.batch_by_collection(collection) or abort(404)
+        querystring = request.args.get('search') or abort(400, 'Param "search" required')
+        match = re.match('^(\w+):(.*)', querystring) or abort(400, 'Invalid search string')
```

Comment:
This isn't blocking for this release, but just note that \w will need to be \\w when we move to support Python 3.12.

Reply:
Noted. https://github.com/dag-hammarskjold-library/dlx-rest/pull/1383/commits/e0b3cc73d93e3e669e40ca1102af8f656ebc0b7d